### ISSUE TITLE
Terminal in side panel layout issues

### ISSUE THREAD

**Issue by ednl** [microsoft/vscode#139457]
Issue Type: <b>Bug</b>

Steps: open VS Code; select View > Appearance Move Panel Right; produce some output in the terminal; important step: close the panel; close VS Code; reopen VS Code; open the panel.

Expected: the terminal should again use the whole side/vertical panel and (preferably) the output should be restored correctly.

Actual: terminal height in the panel is limited to a number of rows at the bottom, as if the whole panel had opened at the bottom. New output will be capped on those rows. Old restored output may be scrambled (width also not consistent and/or not restored). See screenshot.

VS Code version: Code 1.63.2 (Universal) (899d46d82c4c95423fb7e10e68eba52050e30ba3, 2021-12-15T09:37:28.172Z)
OS version: Darwin x64 20.6.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-4250U CPU @ 1.30GHz (4 x 1300)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 4, 4|
|Memory (System)|4.00GB (0.03GB free)|
|Process Argv|--disable-extensions --crash-reporter-id 560275b2-6aa5-4728-9999-6ecd84b106be|
|Screen Reader|no|
|VM|0%|
</details>Extensions disabled

**Comment by ednl** [comment]
![panel-right-1](https://user-images.githubusercontent.com/9798500/146667744-3d817101-4895-4825-a53e-80fe61c82d48.png)
![panel-right-2](https://user-images.githubusercontent.com/9798500/146667751-507091cb-bb6d-4eee-b908-998cd6f799fb.png)

**Comment by Tyriar** [comment]
Struggling to repro this, any more info on how to make it happen? Does it happen every time?
![recording (3)](https://user-images.githubusercontent.com/2193314/146815660-35cc6e76-5e94-4db4-85d9-2badc3346d26.gif)

**Comment by ednl** [comment]
I'm struggling now too! Argh. I confirmed multiple times before I submitted, it happened every time. I see two differences: I use the general release not Insiders, and I stopped the app (Cmd-Q) instead of just closing the window (red button/Cmd-W). So that's what I tried immediately, too: Cmd-W, and now the viewport is correct every time ... No idea if one is directly caused by the other. One thing that does go wrong is after closing the window (Cmd-W) and opening a new window, the panel will always open at the bottom.

Is it possible that height or width are not saved in settings permanently until a certain, very specific condition is met? Like a move of the panel between bottom and right while it is visible, or closing the window while it is visible, or something like that. I tried on a different Mac (M1 instead of x86 from where I opened the issue) where I still get the bug as originally subscribed, see attached video below.

Actions in the video: start VSC, open panel with Ctrl-` => it's clipped, close panel, stop VSC (Cmd-Q), restart VSC (notice no highlight under the icon, so fresh start), open terminal Ctrl-` => clipped, close window (Cmd-W), open new window (notice highlight under icon, so app still running), open terminal Ctrl-` => opens at bottom but not clipped, move to right, still not clipped, close panel, stop VSC, restart VSC, open terminal Ctrl-` => clipped again!

https://user-images.githubusercontent.com/9798500/146935486-da28f7fa-6907-43d3-9b3d-8860858f2991.mov

**Comment by ednl** [comment]
One more detail: this first happened for me in 1.63.0 (general release).

**Comment by Tyriar** [comment]
> I use the general release not Insiders, and I stopped the app (Cmd-Q) instead of just closing the window (red button/Cmd-W)

The repro I did should be the same when `"terminal.integrated.persistentSessionReviveProcess": "onExitAndWindowClose"` is set.

> Is it possible that height or width are not saved in settings permanently until a certain, very specific condition is met?

We store the dimensions of the terminal as part of the session restore, so it should happen like this:

- Resize to stored dimensions
- Print stored data
- Resize to natural dimensions
- Create process

I don't think it's possible to store the buffer without its dimensions

> I tried on a different Mac (M1 instead of x86 from where I opened the issue) where I still get the bug as originally subscribed, see attached video below.

Looks like the side panel is getting some cached bottom panel dimensions or something 🤔, let's track that here as I don't think I'll be able to reproduce the weird 2 character thing.

**Comment by ednl** [comment]
> Looks like the side panel is getting some cached bottom panel dimensions or something 🤔

Yep, that was also my main observation under "Actual" in the OP.

**Comment by svenefftinge** [comment]
I can reliably reproduce this in Gitpod, when starting a fresh workspace and let it open in the background (no focus on the browser).
